### PR TITLE
Ask clients how they would like to be interviewed

### PR DESCRIPTION
**Pull request by adarsh**
Reason for Change
=================
* We should ask clients how they would like to be interviewed by the county.

Change
======
* Create a "Preferences" subsection
* Add a step for interview preferences.

Minor
=====
* Add a note in the `README.md` on skipping around the steps.
* Remove migration command related to paperclip. After removing paperclip, the migration using `add_attachment` no longer works. The suggestion [here][1] was to just rename the columns in the migration.

[1]: https://github.com/thoughtbot/paperclip/issues/1351#issuecomment-29522904

### CODE REVIEW

**Comment by adarsh** [comment]
[As an applicant, I want to be able to choose whether to conduct my required interview over the phone or in-person.](https://trello.com/c/06zqHr3P/133-as-an-applicant-i-want-to-be-able-to-choose-whether-to-conduct-my-required-interview-over-the-phone-or-in-person)

**Comment by adarsh** [comment]
image
image

File: README.md [hunk 1]
```diff
@@ -73,6 +73,16 @@ Ask the team for access to the following services to begin contributing:
 
 ## What's Included
 
+### Step Skipping
+
+This application is a long questionaire.
+You will probably want to work on parts of it
+without completing the whole application.
+
+After booting the server,
```
Comment (jayroh) [inline README.md]: `After booting the server and filling out the first step`

File: README.md [hunk 2]
```diff
@@ -73,6 +73,16 @@ Ask the team for access to the following services to begin contributing:
...
+go to `http://localhost:3000/steps`
+to jump around.
```
Comment (jayroh) [inline README.md]: /gif house of pain

File: app/controllers/preferences_interview_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class PreferencesInterviewController < StandardStepsController
+  def self.step_class
+    PreferencesInterview
+  end
```
Comment (jayroh) [inline app/controllers/preferences_interview_controller.rb]: I could be wrong but I think it'll infer this without needing to explicitly define it?

File: db/migrate/20170323225139_add_documents.rb [hunk 4]
```diff
@@ -7,7 +7,10 @@ def up
       t.timestamps
     end
 
-    add_attachment :documents, :file
+    add_column :documents, :documents_file_name, :string
+    add_column :documents, :documents_content_type, :string
+    add_column :documents, :documents_file_size, :integer
+    add_column :documents, :documents_updated_at, :datetime
```
Comment (jayroh) [inline db/migrate/20170323225139_add_documents.rb]: I think you'll have to add the `documents` table again if we're going to break this out from the []String column in `snap_applications`. This is an old migration that ends up getting dropped in `db/migrate/20170728213954_drop_unused_database_tables.rb`

I'd recommend not doing *this* part until that can all be whipped into shape in one atomic commit.

File: app/controllers/preferences_interview_controller.rb (hunk 3 above)
Comment (jessieay) [inline app/controllers/preferences_interview_controller.rb]: yes! this works as long as the step does not end with an `s`

File: app/views/preferences_interview/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      </div>
+    </div>
+  </header>
+
+  <div class="form-card__content">
+    <%= form_for @step, as: :step, builder: MbFormBuilder, url: current_path, method: :put do |f| %>
+      <%= f.mb_radio_set :interview_preference,
+        "What do you prefer?",
+        [
+          { value: "telephone", label: "Telephone Interview" },
```
Comment (jessieay) [inline app/views/preferences_interview/edit.html.erb]: I feel like we've been making these `value` values symbols elsewhere. they get saved as strings in the DB so doesn't really matter either way, but would be more consistent to go symbol

File: spec/controllers/preferences_interview_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  let(:step) { assigns(:step) }
+  let(:invalid_params) { { step: { interview_preference: "telepathy" } } }
+  let(:step_class) { PreferencesInterview }
+
+  before { session[:snap_application_id] = current_app.id }
+
+  include_examples "step controller"
+
+  describe "#edit" do
+    it "assigns the attributes to the step" do
```
Comment (jessieay) [inline spec/controllers/preferences_interview_controller_spec.rb]: this would be for a separate PR but I think it would be cool to get something like this into the shared example group

File: spec/controllers/preferences_interview_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  describe "#update" do
+    context "the preference is for a telephone interview" do
+      it "is valid" do
+        params = { interview_preference: "telephone" }
+
+        put :update, params: { step: params }
+
+        expect(response).to redirect_to("/steps/legal-agreement")
```
Comment (jessieay) [inline spec/controllers/preferences_interview_controller_spec.rb]: also test that the preference is updated in the DB?

File: db/migrate/20170323225139_add_documents.rb (hunk 4 above)
Comment (adarsh) [inline db/migrate/20170323225139_add_documents.rb]: Yeah I was conflicted about how to approach this. I'll revert the commit for now.

File: spec/controllers/preferences_interview_controller_spec.rb (hunk 6 above)
Comment (adarsh) [inline spec/controllers/preferences_interview_controller_spec.rb]: Yeah I agree!
